packages/discovery/src/index.ts

```diff
@@ -0,0 +1,56 @@
+export type ProviderDetail = {
+  _tag: string
+  info: ProviderInfo
+  provider: Promise<unknown>
```

Comment:
Does this have to be a promise? I would have thought that, by the time the extension responds to the event, it already can provide a well defined interface and doesn't need to give a Promise back?

Reply:
I think so because there is no top level await.

Currently we create the provider by invoking `getLightClientProvider` which returns a Promise. And this promise would resolve well before `requestProvider` is invoked, but theres no way to unwrap that promise since the callback is synchronous. 

https://github.com/paritytech/substrate-connect/blob/main/projects/wallet-template/src/inpage/index.ts#L43

Reply:
Alternatively the extension could create a wrapper and for each method do `providerPromise.then(provider) => provider.invokeInterfaceMethod`, assuming all the methods are async.

Reply:
Mmm that was my thinking! I don't know why that thing is async offhand, and fundamentally I don't really see why the extension wouldn't be able to synchronously provide an interface back (even if it is an interface containing async methods).

So I guess I think that, all else aside, we shouldn't _force_ the interface to be `Promise<..>` and instead let the specific things decide what they return